From 56d307993c0a661c73fcdd72b1392c3719a0c297 Mon Sep 17 00:00:00 2001
Message-Id: <56d307993c0a661c73fcdd72b1392c3719a0c297.1334369310.git.paul.eggleton@linux.intel.com>
In-Reply-To: <cover.1334369310.git.paul.eggleton@linux.intel.com>
References: <cover.1334369310.git.paul.eggleton@linux.intel.com>
From: Joshua Lock <josh@linux.intel.com>
Date: Tue, 10 Apr 2012 17:54:56 -0700
Subject: [PATCH 1/6] Monitor the Manager's State property

Monitor the Manager's State property and update global_ready
appropriately when it changes.

Without this change using the applet with connman 0.79 and
starting the applet after the daemon no status icon is shown.

With this change this icon displays an appropriate state when
the applet launches.

Upstream-Status: Submitted

Signed-off-by: Joshua Lock <josh@linux.intel.com>
---
 applet/main.c |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/applet/main.c b/applet/main.c
index 224e2fd..68a77b1 100644
--- a/applet/main.c
+++ b/applet/main.c
@@ -111,6 +111,16 @@ static void manager_property_changed(DBusGProxy *proxy, const char *property,
 						iterate_list, &path);
 		update_service(proxy, path);
 		g_free(path);
+	} else if (g_str_equal(property, "State") == TRUE) {
+		const gchar *state = g_value_get_string(value);
+
+	  	if (g_strcmp0(state, "ready") == 0 || g_strcmp0(state, "online") == 0) {
+	  		global_ready = TRUE;
+			status_ready(global_strength);
+		} else {
+	  		global_ready = FALSE;
+			status_offline();
+		}
 	}
 }
 
-- 
1.7.5.4

